### DISCUSSION TITLE
'Update' object has no attribute 'reply_to_message'

### DISCUSSION THREAD

**Discussion by raj-patra** [python-telegram-bot/python-telegram-bot#2589] [Q&A] [answered]
Hey,

I am trying to develop a chat bot with ```python-telegram-bot```. 

Let's say, one of it's functionalities include extracting the username of the user who sent the message and the username of the user whose message has been replied to with the command. 

According to the docs, once our app receives an ```Update``` object it should contain the attribute ```reply_to_message``` which would make the task as easy as the snippet below:

```python
# The callback which will handle the command
def extract(self, update, context):
    if update.message.reply_to_message.from_user.username == 'bot_username':
        update.message.reply_text("Something")
    else:
        target = update.message.reply_to_message.from_user.username
        user= update.message.from_user.username
```
If I am not misinformed, this should work if we have the ```\setprivacy``` setting turned off from Bot Father, but the ```Update``` object only includes the attribute ```reply_to_message``` if I reply to any of the messages sent by the bot and not for replies to regular users.

Where am I going wrong? Any help is greatly appreciated. Thanks.

**Comment by raj-patra** [comment, marked as answer]
Never mind.

I got where I lacked. Apparently, if I try to reply to a user's message in a group which was sent before the bot's privacy setting was turned off, I would not get the ```reply_to_message``` attribute in the ```Update``` object.

Funny, how I actually had to start a Q&A to finally understand what was happening. :)

**Reply by Poolitzer** [reply]
Telegram is not very good in documentation. You fell right into its trap :D
